web/src/components/Diagram/components/DAG/SpanNode.tsx

```diff
@@ -0,0 +1,30 @@
+import Text from 'antd/lib/typography/Text';
+import {capitalize} from 'lodash';
+import {Handle, NodeProps, Position} from 'react-flow-renderer';
+
+import {SemanticGroupNamesToText} from 'constants/SemanticGroupNames.constants';
+import {INodeDataSpan} from 'types/DAG.types';
+import * as S from './SpanNode.styled';
+
+interface IProps extends NodeProps<INodeDataSpan> {}
+
+const SpanNode = ({data, id, selected}: IProps) => {
```

Comment:
Refactored the `TraceNode` component into this new `SpanNode`. Removed the "component factory" for now. I don't think we need the factory today. We can add it in the future if needed.